packages/@react-aria/overlays/test/useModal.test.js

```diff
@@ -88,4 +88,102 @@ describe('useModal', function () {
+      </div>
+    );
+    let rootProvider = res.getByTestId('root-provider');
+
+    expect(rootProvider).not.toHaveAttribute('aria-hidden');
+
+    res.rerender(
+      <div id="alternateContainer" data-testid="alternate-container">
+        <Example showModal container={() => document.getElementById('alternateContainer')} />
+      </div>
+    );
+
+    let modalProvider = res.getByTestId('modal-provider');
+
+    expect(rootProvider).toHaveAttribute('aria-hidden', 'true');
+    expect(modalProvider).not.toHaveAttribute('aria-hidden');
+
+    res.rerender(
+      <div id="alternateContainer" data-testid="alternate-container">
+        <Example showModal container={() => document.getElementById('alternateContainer')}>
+          <Modal providerId="inner-modal-provider" modalId="inner-modal">Inner</Modal>
+        </Example>
+      </div>
+    );
+
+    let innerModalProvider = res.getByTestId('inner-modal-provider');
+
+    expect(rootProvider).toHaveAttribute('aria-hidden', 'true');
+    expect(modalProvider).toHaveAttribute('aria-hidden');
+    expect(innerModalProvider).not.toHaveAttribute('aria-hidden');
+
+    res.rerender(
+      <div id="alternateContainer" data-testid="alternate-container">
+        <Example container={() => document.getElementById('alternateContainer')} />
+      </div>
+    );
+    expect(rootProvider).not.toHaveAttribute('aria-hidden');
+  });
+
+  describe('error state', function () {
+    const consoleError = console.error;
+    beforeEach(() => {
+      console.error = jest.fn();
+    });
+
+    afterEach(() => {
+      console.error = consoleError;
+    });
+    it('if inside another container, throws an error', function () {
+      let res = render(
+        <div id="alternateContainer" data-testid="alternate-container">
+          <Example container={() => document.getElementById('alternateContainer')}>
+            <div id="nestedContainer" />
+          </Example>
+        </div>
+      );
+      let rootProvider = res.getByTestId('root-provider');
+
+      expect(rootProvider).not.toHaveAttribute('aria-hidden');
+
+      res.rerender(
+        <div id="alternateContainer" data-testid="alternate-container">
+          <Example showModal container={() => document.getElementById('alternateContainer')}>
+            <div id="nestedContainer" />
+          </Example>
+        </div>
+      );
+
+      let modalProvider = res.getByTestId('modal-provider');
+
+      expect(rootProvider).toHaveAttribute('aria-hidden', 'true');
+      expect(modalProvider).not.toHaveAttribute('aria-hidden');
+      expect(() =>
+        res.rerender(
+          <div id="alternateContainer" data-testid="alternate-container">
+            <Example showModal container={() => document.getElementById('alternateContainer')}>
+              <div id="nestedContainer" />
+              <Modal
+                container={() => document.getElementById('nestedContainer')}
```

Comment:
Just for clarification, if this pointed to the "alternateContainer" then the modal would be be properly nested correct?

Reply:
right, and we shouldn't see the error in that case